### PR TITLE
Guest and Backward Compatibility Docs

### PR DESCRIPTION
**Pull request by deepakkinni**
Signed-off-by: Deepak Kinni <dkinni@vmware.com>

### CODE REVIEW

File: docs/guest.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#Tanzu Kubernetes Grid Service
```
Comment (lintongj) [inline docs/guest.md:1]: Leave one space between `#` and words

File: docs/backward-compat.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+only on a Vanilla Setup.
+
+Prior to attempting restore or delete backup (<=1.0.2), ensure that a vSphere snapshot location exists prior to attempting a restore of older backup.
+If not, create one.
+
+```bash
+velero snapshot-location create <snapshot location name> --provider velero.io/vsphere
+```
+
+After restore/delete of <=1.0.2 plugin Backup, the vSphere snapshot location can be safely deleted. The vSphere snapshot
```
Comment (xing-yang) [inline docs/backward-compat.md:13]: <=1.0.2 plugin Backup -> backups created using plugin version <= 1.0.2

File: docs/backward-compat.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+location is only necessary when attempting to restore or delete a older(<=1.0.2) plugin backup.
```
Comment (xing-yang) [inline docs/backward-compat.md:14]: a older(<=1.0.2) plugin backup -> a backup from an older plugin (<=1.0.2)
Same comment applies to other places as well.

File: docs/guest.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
+# Tanzu Kubernetes Grid Service
+This document discusses the velero vSphere plugin installation process in a Tanzu Kubernetes Grid Service environment
+aka as Guest Cluster. For more information on Tanzu Kubernetes Grid Service refer to their [documentation](https://docs.vmware.com/en/VMware-Tanzu-Kubernetes-Grid/index.html).
+
+## Compatibility
+| vSphere/ESXi Version       | vSphere CSI Version                                                | Kubernetes Version | Velero Version    | Velero Plugin for vSphere Version |
+|----------------------------|--------------------------------------------------------------------|--------------------|-------------------|-----------------------------------|
+| vSphere 7.0P02/ESXi 7.0P02 | Driver version is not v2.1.0, but feature is equivalent to v2.1.0. | v1.16-v1.19        | v1.5.1 and higher | v1.1.0 and higher                 |
```
Comment (xing-yang) [inline docs/guest.md:8]: Driver version is not v2.1.0, but feature is equivalent to v2.1.0. -> CSI driver bundled with TKGS

File: docs/guest.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+## Prerequisites
+* Prior to the installation of Velero vSphere plugin in Guest cluster it is necessary to have Velero and Velero 
+vSphere plugin installed in the Supervisor Cluster, please refer to [Supervisor Cluster Velero vSphere Plugin](docs/supervisor.md)
```
Comment (xing-yang) [inline docs/guest.md:12]: This doc link is broken.

All doc links pointing to our own docs are broken.

File: docs/guest.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+installation guide.
+* DataManager needs to be installed, please refer [DataManager Documentation](docs/supervisor-datamgr.md)
+
+## Install
+Please refer [Vanilla Installation Guide](docs/vanilla.md)
```
Comment (xing-yang) [inline docs/guest.md:17]: Should not refer to a Vanilla guide.  The install part should be in a separate doc.

File: docs/guest.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+* The installation of Velero vSphere Plugin in guest cluster may take over ten minutes to complete, this is primarily
+because the plugin waits for the velero app operator to write the Secret into a predefined namespace.
+* The installation may hang forever if Velero is not installed in the supervisor cluster.
```
Comment (xing-yang) [inline docs/guest.md:21]: Make the first letter capital: Supervisor

File: docs/guest.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+## Prerequisites
+* Prior to the installation of Velero vSphere plugin in Guest cluster it is necessary to have Velero and Velero 
+vSphere plugin installed in the Supervisor Cluster, please refer to [Supervisor Cluster Velero vSphere Plugin](docs/supervisor.md)
+installation guide.
+* DataManager needs to be installed, please refer [DataManager Documentation](docs/supervisor-datamgr.md)
+
+## Install
+Please refer [Vanilla Installation Guide](docs/vanilla.md)
+
+* The installation of Velero vSphere Plugin in guest cluster may take over ten minutes to complete, this is primarily
```
Comment (xing-yang) [inline docs/guest.md:19]: Guest

File: docs/guest.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+because the plugin waits for the velero app operator to write the Secret into a predefined namespace.
+* The installation may hang forever if Velero is not installed in the supervisor cluster.
+
+## Uninstall
+Please refer [Vanilla Uninstall Guide](docs/vanilla.md)
+
+## Backup
+Please refer [Vanilla Backup Guide](docs/vanilla.md)
+
+On Backup, a Snapshot CRD is created to track the vSphere volume snapshot operation and data upload.
```
Comment (xing-yang) [inline docs/guest.md:29]: Add an example snapshot CRD.  Can be added in a followup PR later.

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/427/

File: docs/guest.md (hunk 4 above)
Comment (xing-yang) [inline docs/guest.md:8]: Looks like they call the new release  vSphere 7.0 U1c / P02

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/429/

File: docs/guest.md (hunk 9 above)
Comment (deepakkinni) [reply docs/guest.md]: Done, resolved other comments as well.

**Review by xing-yang — APPROVED** [review]
LGTM

**Comment by svcbot-qecnsdp** [comment]
Congratulations !! 

All the pre-check tests passed
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/433/

**Pull request merged**